packages/dci-api/src/crypto/encrypt.ts

```diff
@@ -0,0 +1,78 @@
+import { FlattenedEncrypt, type FlattenedJWE, importJWK } from 'jose'
+import { type components } from '../registry-core-api'
+import { z } from 'zod'
+import { ValidationError, fromZodError } from 'zod-validation-error'
+
+function jweToEncryptedMessage(
+  jwe: FlattenedJWE
+): components['schemas']['EncryptedMessage'] {
+  if (jwe.header?.alg === undefined) {
+    throw new Error('Missing required header paramater "alg" after encrypting')
+  }
+  if (jwe.header.enc === undefined) {
+    throw new Error('Missing required header paramater "enc" after encrypting')
+  }
+  if (jwe.header.kid === undefined) {
+    throw new Error('Missing required header paramater "kid" after encrypting')
+  }
+  if (jwe.encrypted_key === undefined) {
+    throw new Error('encrypted_key is required after encrypting')
+  }
```

Comment:
In which cases would these be missing? Can it happen

Reply:
They shouldn't be, I was mainly doing this to make typescript happy. :sweat_smile: